### PR TITLE
Add tslint config

### PR DESCRIPTION
**Pull request by jalack**
Adding TSLint config that we've been using in a few of our repositories. Open to any and all suggestions.

See https://palantir.github.io/tslint/2016/03/31/sharable-configurations-rules.html & https://palantir.github.io/tslint/ for more info on TSLint.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
   },
   "homepage": "https://github.com/mixpanel/eslint-config",
   "peerDependencies": {
-    "eslint": ">=3.1.0 <4.0.0"
+    "eslint": ">=3.1.0 <4.0.0",
+    "tslint": ">=5.0.0"
```
Comment (tdumitrescu) [inline package.json]: Isn't this a little dangerous, allowing any major version bump indiscriminately?

File: tslint-config.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  extends: `tslint:recommended`,
+  defaultSeverity: `error`,
+  rules: {
+    'array-type': [true, `generic`],
+    'arrow-parens': false,
+    'arrow-return-shorthand': true,
+    'import-spacing': true,
+    'indent': [true, `spaces`],
+    'interface-name': [true, `never-prefix`],
+    'max-line-length': [true, 140],
```
Comment (nojvek) [inline tslint-config.js:11]: I'm personally not a big fan of max-line-length rule. I let the IDE handle the formatting of long lines and wrapping them around. But I'm okay with it if you feel strongly.

**Review by nojvek — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (jalack) [reply package.json]: good call, will keep a major version restriction. while we're here, do you see any reason not to bump up peer dependency eslint version to `<5.0.0`? nothing jumping out at me here: https://eslint.org/docs/user-guide/migrating-to-4.0.0#eslint-recommended-changes

File: tslint-config.js (hunk 2 above)
Comment (jalack) [reply tslint-config.js:11]: I generally hate it when there are aggressive restrictions, though I think some sanity check is worthwhile, and 140 seems about right to me. if you feel strongly I can remove.

**Comment by tdumitrescu** [comment]
Published in 3.0.0 (major version bump because of the eslint major version bump)

**Pull request merged**